src/hiveboard_bridge/src/ReceiveAction.cpp

```diff
@@ -0,0 +1,16 @@
+#include "hiveboard_bridge/ReceiveAction.h"
+
+ReceiveAction::ReceiveAction(IHiveMindHostDeserializer& deserializer,
+                             IMessageHandler& messageHandler) :
+    m_deserializer(deserializer), m_messageHandler(messageHandler) {}
+
+void ReceiveAction::doAction() {
```

Comment:
The naming is a bit confusing here, it seems to be "Do/call this action if a message is received" While it's actually "I fetch a message myself and if there is one, i'll do it". 

You could pass the message to the doAction function, decoupling the fetching/receiving of the message and the handling of the message and would match the naming. 

Or you could even just put this code directly in the thread  wrapper since it's really just a check for the message (If you do, you could rename it MessageThread or something similar)

Reply:
At first, this was directly in the ThreadWrapper, but I wanted to have a way to test it easily (without the need to start a thread or test a private method) so I put it in a separate class. I'll rename `doAction` to `fetchAndProcessMessage`.